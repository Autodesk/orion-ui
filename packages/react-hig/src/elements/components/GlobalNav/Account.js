/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/

import * as PropTypes from 'prop-types';
import HIGElement from '../../HIGElement';
import createComponent from '../../../adapters/createComponent';

export class Account extends HIGElement {
  commitUpdate(updatePayload, oldProps, newProp) {
    const mapping = {
      image: 'setImage',
      label: 'setLabel'
    };

    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case mapping[propKey]:
          this.hig[mapping[propKey]](propValue);
          break;
        case 'active':
          if (propValue) {
            this.hig.activate();
          } else {
            this.hig.deactivate();
          }
          break;          
        default:
          this.commitPropChange(propKey, propValue);
      }
    }
  }

  componentDidMount(){
    if (this.initialProps.activatedAccount == true ) {
      
    }
  }
}

const AccountComponent = createComponent(Account);

AccountComponent.propTypes = {
  image: PropTypes.string,
  label: PropTypes.string,
  activate: PropTypes.func,
  deactivate: PropTypes.func,
  onClick: PropTypes.func
};

AccountComponent.__docgenInfo = {
  props: {
    label: {
      description: 'sets {String} the label displayed for an account in Account/account switcher'
    },
    image: {
      description: 'sets {String} the image displayed for an account in Account/account switcher'
    },
    activate: {
      description: '{func} activates the Account'
    },
    deactivate: {
      description: '{func} deactivates the Account'
    },
    onclick: {
      description: '{func} calls the provided callback when user clicks on the Account'
    }
  }
};

export default AccountComponent;
