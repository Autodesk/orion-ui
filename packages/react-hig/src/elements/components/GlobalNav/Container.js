/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

export class Container extends HIGElement {
  componentDidMount() {
    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.mount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.mount();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance) {
    if (instance instanceof TopNav) {
      if (this.mounted) {
        this.hig.addTopNav(instance.hig);
        instance.mount();
      } else {
        this.topNav = instance;
      }
    } else if (instance instanceof SubNav) {
      if (this.mounted) {
        this.hig.addSubNav(instance.hig);
        instance.mount();
      } else {
        this.subNav = instance;
      }
    } else {
      throw new Error(`unknown type`);
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    instance.unmount();
  }
}

const ContainerComponent = createComponent(Container);

ContainerComponent.propTypes = {
  children: HIGChildValidator([TopNavComponent, SubNavComponent, Slot])
};

ContainerComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding TopNav, SubNav, and Slot'
    }
  }
};

ContainerComponent.TopNav = TopNavComponent;
ContainerComponent.SubNav = SubNavComponent;
ContainerComponent.Slot = Slot;

export default ContainerComponent;
