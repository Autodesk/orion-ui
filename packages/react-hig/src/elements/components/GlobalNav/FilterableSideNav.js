import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SideNav from '../../../adapters/GlobalNav/SideNav/SideNavAdapter';

const LinkList = SideNav.LinkList;
const Link = SideNav.LinkList.Link;
const Search = SideNav.Search;
const SectionList = SideNav.SectionList;
const Section = SideNav.SectionList.Section;
const SectionCollapse = SideNav.SectionList.Section.SectionCollapse;
const Group = SideNav.SectionList.Section.Group;
const Module = SideNav.SectionList.Section.Group.Module;
const ModuleCollapse = SideNav.SectionList.Section.Group.Module.ModuleCollapse;
const Submodule = SideNav.SectionList.Section.Group.Module.Submodule;

export default class FilterableSideNav extends Component {
  static propTypes = {
    query: PropTypes.string,
    items: PropTypes.shape({
      links: PropTypes.arrayOf(PropTypes.object),
      sections: PropTypes.arrayOf(
        PropTypes.shape({
          groups: PropTypes.arrayOf(
            PropTypes.shape({
              modules: PropTypes.shape({
                submodules: PropTypes.arrayOf(PropTypes.object)
              })
            })
          )
        })
      )
    }).isRequired
  };

  render() {
    return (
      <SideNav>
        {
          <LinkList>
            {this.props.items.links.map((link, i) => {
              return <Link {...link} key={link.title} />;
            })}
          </LinkList>
        }
      </SideNav>
    );
  }
}
