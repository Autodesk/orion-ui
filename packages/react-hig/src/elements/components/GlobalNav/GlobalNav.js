/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import * as HIG from 'hig.web';
import * as PropTypes from 'prop-types';

import HIGElement from '../../HIGElement';
import HIGChildValidator from '../../HIGChildValidator';
import createComponent from '../../../adapters/createComponent';

import SideNavComponent, { SideNav } from './SideNav';
import TopNavComponent, { TopNav } from './TopNav';
import SubNavComponent, { SubNav } from './SubNav';
import Slot from './Slot';

class GlobalNav extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.componentDidMount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.componentDidMount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.componentDidMount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNav:
        return new TopNav(this.hig.partials.TopNav, props);
      case SubNav:
        return new SubNav(this.hig.partials.SubNav, props);
      case SideNav:
        return new SideNav(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNav) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof TopNav) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else if (instance instanceof SubNav) {
      if (this.SubNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.componentDidMount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNav) {
      this.sideNav = null;
    } else if (instance instanceof TopNav) {
      this.topNav = null;
    } else if (instance instanceof SubNav) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          if (propValue) {
            this.hig.showSideNav();
          } else {
            this.hig.hideSideNav();
          }
          break;
        }
        case 'children': {
          /* no-op */
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavComponent = createComponent(GlobalNav);

GlobalNavComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNavComponent,
    TopNavComponent,
    SubNavComponent,
    Slot
  ])
};

GlobalNavComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavComponent.SideNav = SideNavComponent;
GlobalNavComponent.TopNav = TopNavComponent;
GlobalNavComponent.SubNav = SubNavComponent;
GlobalNavComponent.Slot = Slot;

export default GlobalNavComponent;
