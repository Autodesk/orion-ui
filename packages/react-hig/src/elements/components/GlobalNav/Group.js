/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';
import ModuleComponent, { Module } from './Module';

export class Group extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);
    this.modules = new HIGNodeList({
      type: Module,
      HIGConstructor: this.hig.partials.Module,
      onAdd: (instance, beforeInstance) => {
        this.hig.addModule(instance, beforeInstance);
      }
    });

    this.state = {};
    this._render = this._render.bind(this);
  }

  componentDidMount() {
    this.modules.componentDidMount();
    this._render();
  }

  createElement(ElementConstructor, props) {
    return this.modules.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.modules.insertBefore(instance, insertBeforeIndex);
    if (this.mounted) {
      this._render();
    }
  }

  removeChild(instance) {
    this.modules.removeChild(instance);
  }

  commitUpdate(updatePayload, oldProps, newProps) {
    this.processUpdateProps(updatePayload).then(this._render);
  }

  isVisible() {
    return this.state.isVisible;
  }

  expandModules() {
    this.modules.forEach(module => {
      module.commitUpdate({query: undefined})
      module.expandAll();
    });
    this._render();
  }

  collapseModules() {
    this.modules.forEach(module => {
      this.commitUpdate({query: undefined})
      module.collapseAll();
    });
    this._render();
  }

  _render() {
    const matches = this.modules.map(module => {
      module.commitUpdate({
        query: this.props.query,
        activeModule: this.props.activeModule,
        activeSubmodule: this.props.activeSubmodule
      });

      return module.isVisible();
    });

    if (matches.some(m => m)) {
      this.hig.show();
      this.state.isVisible = true;
    } else {
      this.hig.hide();
      this.state.isVisible = false;
    }
  }
}

const GroupComponent = createComponent(Group);

GroupComponent.propTypes = {
  children: HIGChildValidator([ModuleComponent])
};

GroupComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Module'
    }
  }
};

GroupComponent.Module = ModuleComponent;

export default GroupComponent;
