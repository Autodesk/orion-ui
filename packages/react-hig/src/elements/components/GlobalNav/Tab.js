/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import PropTypes from 'prop-types';

export class Tab extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    ['onActiveTab', 'callOnActiveTab'].forEach(fn => {
      this[fn] = this[fn].bind(this);
    });

  }

  componentDidMount() {
    this.hig.onClick(this.callOnActiveTab)
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.processUpdateProps(updatePayload)
      .mapToHIGEventListeners(['onClick'])
      .mapToHIGFunctions({
        label: 'setLabel'
      })
      .handle('active', value => {
        if (value) {
          this.callOnActiveTab();
        }
      })
  }

  callOnActiveTab(){
    if (!this._onActiveTab) {
      return;
    }

    this._onActiveTab(this);
  }

  onActiveTab(callback){
    this._onActiveTab = callback; 
  }
}

const TabComponent = createComponent(Tab);

TabComponent.propTypes = {
  active: PropTypes.bool,
  label: PropTypes.string
};

TabComponent.__docgenInfo = {
  props: {
    active: {
      description: 'activates or deactivates the tab'
    },
    label: {
      description: 'sets the text of a tab'
    }
  }
};

export default TabComponent;
