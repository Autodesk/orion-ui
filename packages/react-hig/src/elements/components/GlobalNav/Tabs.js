/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import createComponent from '../../../adapters/createComponent';
import HIGElement from '../../HIGElement';
import HIGNodeList from '../../HIGNodeList';
import HIGChildValidator from '../../HIGChildValidator';

import TabComponent, { Tab } from './Tab';

// Does not extend HIGElement because it's not a real HIG component
export class Tabs extends HIGElement {
  constructor(HigContructor, initialProps) {
    super(HigContructor, initialProps);
    this.tabs = new HIGNodeList({
      type: Tab,
      HIGConstructor: this.hig.partials.Tab,
      onAdd: (instance, beforeInstance) => {
        this.hig.addTab(instance, beforeInstance);
      }
    });
    this.props = initialProps;
    this.state = {};

    ['setActiveTab'].forEach(fn => {
      this[fn] = this[fn].bind(this)
    });
  }

  componentDidMount() {
    this.tabs.componentDidMount();
    this._render();
  }

  createElement(ElementConstructor, props) {
    return this.tabs.createElement(ElementConstructor, props);
  }

  insertBefore(instance, insertBeforeIndex) {
    this.tabs.insertBefore(instance, insertBeforeIndex);
    instance.onActiveTab(this.setActiveTab);
    if (this.state.activeTab === undefined){
      this.state.activeTab = instance
    }
  }

  removeChild(instance) {
    this.tabs.removeChild(instance);
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    this.props = { ...this.props, ...updatePayload };
  }
  
  setActiveTab(tab){
    this.state.activeTab = tab;
    if (this.props.onTabChange) {
      this.props.onTabChange(tab);
    }
    this._render();
  };

  _render(){
    this.tabs.forEach(tab => {
      this.state.activeTab === tab ? tab.hig.activate() : tab.hig.deactivate();
    })
  }
}

const TabsComponent = createComponent(Tabs);

TabsComponent.propTypes = {
  children: HIGChildValidator([TabComponent])
};

TabsComponent.__docgenInfo = {
  props: {
    children: {
      description: 'support adding Tab components'
    }
  }
};

TabsComponent.Tab = TabComponent;

export default TabsComponent;
