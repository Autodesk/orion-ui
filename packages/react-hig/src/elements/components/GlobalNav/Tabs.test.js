/**
Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

*/
import { mount } from 'enzyme';
import * as HIG from 'hig.web';
import React from 'react';

import GlobalNav from './GlobalNav';
import Tabs from './StatefulTabs';
import Tab from '../../../adapters/Tab';

const Context = props => {  
  return (
    <GlobalNav>
      <GlobalNav.SubNav>
        <Tabs defaultSelectedTab={props.tabs[0].id}>
          {props.tabs.map(tab => <Tabs.Tab key={tab.key} label={tab.label} id={tab.id} />)}
        </Tabs>
      </GlobalNav.SubNav>
    </GlobalNav>
  );
};

describe('<Tabs>', () => {
  it('renders tabs', () => {
    const tabs = [{ key: 7, label: 'Hello', id: 7}, { key: 8, label: 'World', id: 9} ];
    const reactContainer = document.createElement('div');
    const wrapper = mount(<Context tabs={tabs} />, {
      attachTo: reactContainer
    });

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
  });

  it('adds tabs', () => {
    const initialTabs = [ { key: 1, label: 'Hello', id: 1},{ key: 2, label: 'World', id: 2 }];
    const updatedTabs = 
    [
      { key: 3, label: 'Hello', id: 3 }, 
      { key: 4, label:'There', id: 4 }, { key: 5, label: 'World', id: 5}, 
      { key: 6, label: 'Yes', id: 6}];
    const reactContainer = document.createElement('div');
    const wrapper = mount(<Context tabs={initialTabs} />, {
      attachTo: reactContainer
    });

    wrapper.setProps({ tabs: updatedTabs });

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
  });

  it('removes tabs', () => {
    const initialTabs = [{key: 10, label: 'Hello', id: 10}, {key: 11, label: 'World', id: 11}];
    const updatedTabs = [{key: 12, label: 'Updated Tabs', id: 12}];
    const reactContainer = document.createElement('div');
    const wrapper = mount(<Context tabs={initialTabs} />, {
      attachTo: reactContainer
    });

    wrapper.setProps({ tabs: updatedTabs });

    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();
  });
});
