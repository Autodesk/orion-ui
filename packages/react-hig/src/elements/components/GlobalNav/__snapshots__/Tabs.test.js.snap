// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs> adds tabs 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
    <!--TOPNAV-->
    <div class=\\"hig__global-nav__container__sub-nav\\">
    <div class=\\"hig__global-nav__container__sub-nav__left__icon\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left\\">Module Name</div>
    <div class=\\"hig__global-nav__sub-nav__tabs\\">
    <div class=\\"hig__global-nav__sub-nav__tabs__tab\\">Well</div><div class=\\"hig__global-nav__sub-nav__tabs__tab\\">Hello</div><div class=\\"hig__global-nav__sub-nav__tabs__tab\\">World</div><div class=\\"hig__global-nav__sub-nav__tabs__tab\\">!</div><!--TAB-->
</div><!--TABS-->
    <div class=\\"hig__global-nav__container__sub-nav__left__icon hig__global-nav__container__sub-nav__spacer\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left hig__global-nav__container__sub-nav__spacer\\">Module Name</div>
</div><!--SUBNAV-->
    <div class=\\"hig__global-nav__container__slot\\">
        <!--SLOT-->
    </div>
</div><!--CONTAINER-->
</div>"
`;

exports[`<Tabs> removes tabs 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
    <!--TOPNAV-->
    <div class=\\"hig__global-nav__container__sub-nav\\">
    <div class=\\"hig__global-nav__container__sub-nav__left__icon\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left\\">Module Name</div>
    <div class=\\"hig__global-nav__sub-nav__tabs\\">
    <div class=\\"hig__global-nav__sub-nav__tabs__tab\\">World</div><!--TAB-->
</div><!--TABS-->
    <div class=\\"hig__global-nav__container__sub-nav__left__icon hig__global-nav__container__sub-nav__spacer\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left hig__global-nav__container__sub-nav__spacer\\">Module Name</div>
</div><!--SUBNAV-->
    <div class=\\"hig__global-nav__container__slot\\">
        <!--SLOT-->
    </div>
</div><!--CONTAINER-->
</div>"
`;

exports[`<Tabs> renders tabs 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
    <!--TOPNAV-->
    <div class=\\"hig__global-nav__container__sub-nav\\">
    <div class=\\"hig__global-nav__container__sub-nav__left__icon\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left\\">Module Name</div>
    <div class=\\"hig__global-nav__sub-nav__tabs\\">
    <div class=\\"hig__global-nav__sub-nav__tabs__tab\\">Hello</div><div class=\\"hig__global-nav__sub-nav__tabs__tab\\">World</div><!--TAB-->
</div><!--TABS-->
    <div class=\\"hig__global-nav__container__sub-nav__left__icon hig__global-nav__container__sub-nav__spacer\\">#</div>
    <div class=\\"hig__global-nav__container__sub-nav__left hig__global-nav__container__sub-nav__spacer\\">Module Name</div>
</div><!--SUBNAV-->
    <div class=\\"hig__global-nav__container__slot\\">
        <!--SLOT-->
    </div>
</div><!--CONTAINER-->
</div>"
`;
